lib/src/easy_localization_app.dart

```diff
@@ -0,0 +1,168 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/widgets.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import 'localization.dart';
+
+// :( see line 55
+Locale _savedLocale;
+
+class _EasyLocalizationProvider extends InheritedWidget {
+  _EasyLocalizationProvider({Key key, this.child, this.data})
+      : super(key: key, child: child);
+  final _EasyLocalizationState data;
+  final Widget child;
+
+  static _EasyLocalizationProvider of(BuildContext context) {
+    return context
+        .dependOnInheritedWidgetOfExactType<_EasyLocalizationProvider>();
+  }
+
+  @override
+  bool updateShouldNotify(_EasyLocalizationProvider oldWidget) => true;
+}
+
+class EasyLocalization extends StatefulWidget {
+  final Widget child;
+  final List<Locale> supportedLocales;
+  final Locale fallbackLocale;
+  final _EasyLocalizationDelegate delegate;
+  final bool useOnlyLangCode;
+  final String path;
+  final AssetLoader assetLoader;
+
+  EasyLocalization({
+    @required this.child,
+    @required this.supportedLocales,
+    @required this.fallbackLocale,
+    @required this.path,
+    this.useOnlyLangCode = false,
+    this.assetLoader = const RootBundleAssetLoader(),
+  })  : assert(supportedLocales.contains(fallbackLocale)),
+        delegate = _EasyLocalizationDelegate(
+            path: path, supportedLocales: supportedLocales);
+
+  _EasyLocalizationState createState() => _EasyLocalizationState();
+
+  static _EasyLocalizationState of(BuildContext context) =>
+      _EasyLocalizationProvider.of(context).data;
+
+  // that is a hack :P
+  // @TODO is there a better way to initialize the saved locale so that the app
+  // initializes with the last choosen locale.
+  // Right now developer needs to call this in main and if a saved locale is found
+  // we save it on a global variable in this field.
+  static loadSavedLocale() async {
+    SharedPreferences _preferences = await SharedPreferences.getInstance();
+    var _codeLang = _preferences.getString('codeLa');
+    var _codeCoun = _preferences.getString('codeCa');
+
+    if (_codeLang != null) _savedLocale = Locale(_codeLang, _codeCoun);
+  }
+}
```

Comment:
`loadSavedLocale()` this should be privete

Reply:
I have changed this check the new impl.